x/gamm/types/account.go

```diff
@@ -32,7 +32,7 @@ type PoolAccountI interface {
 	GetPoolAssets(denoms ...string) ([]PoolAsset, error)
 	SetPoolAssets(assets []PoolAsset) error
 	GetAllPoolAssets() []PoolAsset
-	SetTokenWeight(denom string, weight sdk.Int) error
+	PokeTokenWeights(blockTime time.Time)
```

Comment:
Why not keep UpdateWeights as a function on the interface?

Reply:
It doesn't belong on the interface? Nothing external should ever call that

Reply:
Oh right, we said governance should only update weights by creating a smooth change process